src/app/components/Tokens/TokenHolders.tsx

```diff
@@ -0,0 +1,113 @@
+import { FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Table, TableCellAlign, TableColProps } from '../Table'
+import { BareTokenHolder } from '../../../oasis-nexus/api'
+import { TablePaginationProps } from '../Table/TablePagination'
+import { AccountLink } from '../Account/AccountLink'
+import { fromBaseUnits } from '../../utils/helpers'
+import { TokenPriceInfo } from '../../../coin-gecko/api'
+
+type TableTokenHolder = BareTokenHolder & {
+  markAsNew?: boolean
+}
+
+type TokenHoldersProps = {
+  holders?: TableTokenHolder[]
+  isLoading: boolean
+  decimals: number | undefined
+  totalSupply: string | undefined
+  limit: number
+  tokenPrice?: TokenPriceInfo
+  pagination: false | TablePaginationProps
+}
+
+export const TokenHolders: FC<TokenHoldersProps> = ({
+  isLoading,
+  limit,
+  pagination,
+  holders,
+  decimals = 18,
```

Comment:
It does not match all ParaTimes. Can we remove this default?

Reply:
The problem is that the decimals field is [optional](https://github.com/oasisprotocol/nexus/blob/main/api/spec/v1.yaml#L2468) on tokens. So I guess in case it's missing, we need to look at the paratime config....

Reply:
Done